### ISSUE TITLE
Provide installation stats for grains

### ISSUE THREAD

**Issue by mitar** [sandstorm-io/sandstorm#1954]
I am not completely sure if I just missed this, but as an app author it would be great to have some stats about how app is being used available. This is one of advantages of app markets provide to authors.

For example, I would love to know how many times my app has been installed, how many grains have been created, through time. Maybe even a histogram for how long grains were around (so that I can see if most of users just run it shortly and destroy it, or keep it around). Also which version of grains people have (are they updating it, or not). In general, some stats about life-cycles of grains.

As people can install their own Sandstorm servers, I think admin could decide at installation time if they are OK submitting anonymous usage data for grains to the app store from which the app was downloaded.

Alternatively, app's definition could provide some URL to which stats should be submitted.

**Comment by ocdtrekkie** [comment]
I believe Sandstorm already does collect some usage data if the server admin opts in, though I'm not sure exactly what's collected right now.

I would take huge issue with individual Sandstorm servers reporting stats back to app developers directly, as it would identify the source of each stat broadcast, and for single-user servers, it wouldn't be particularly "anonymized". I'd say reporting statistics back to app authors is almost what Sandstorm is best at _preventing_.

**Comment by mitar** [comment]
> I'd say reporting statistics back to app authors is almost what Sandstorm is best at preventing.

Yes, so there should be some official channels which would allow/provide it. This is really valuable info for app developers. So I am thinking here about non-malicious app developers trying to do the best possible for users.

We should of course want to prevent misuses. But this can also be a community effort to moderate such things. For example, maybe stats reporting could be allowed only for open source apps which one can audit.

**Comment by kentonv** [comment]
Probably the right answer here is that we should share _aggregate_ stats collected by Sandstorm back to app authors. You couldn't tell whose server was running what, but you could get a count of how many active users you have and such. Of course, only servers which opted into stats collection would be counted.

**Comment by paulproteus** [comment]
I'm +1 on this; I would say it's merely a matter of timing and priorities.
